### PR TITLE
Update README.md to reflect changes made to release artifacts and installation instructions

### PR DESCRIPTION
**Pull request by inteon**
These changes reflect the new processes starting from v0.6.0: https://github.com/cert-manager/openshift-routes/releases/tag/v0.6.0

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    - Build and publish any container images
+    - Build and bundle the Helm chart
+    - Create a draft GitHub release
+    - Upload the Helm chart tarball to the GitHub release
+3. Once the draft GitHub release has been created, download and test the resulting Helm chart.
+4. Create a PR in the [jetstack/jetstack-charts repository on GitHub](https://github.com/jetstack/jetstack-charts), containing the Helm chart file that is attached to the draft GitHub release. This is only currently possible for maintainers inside Venafi, but will be changed in the future.
+5. Wait for the PR to be merged and verify that the Helm chart is available from https://charts.jetstack.io.
+6. Visit the [releases page], edit the draft release, click "Generate release notes", then edit the notes to add the following to the top
+    ```
+    approver-policy provides a policy engine for certificates issued by cert-manager!
```
Comment (SgtCoDFish) [inline RELEASE.md:29]: suggestion: Copy past mistake!

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Artifacts
+
+This repo will produce the following artifacts each release. For documentation on how those artifacts are produced see the "Process" section.
+
+- *Container Images* - Container images for the are published to . 
+- *Helm chart* - An official Helm chart is maintained within this repo and published to `charts.jetstack.io` on each release.
+
+[release workflow]: https://github.com/cert-manager/approver-policy/actions/workflows/release.yaml
+[releases page]: https://github.com/cert-manager/approver-policy/releases
```
Comment (SgtCoDFish) [inline RELEASE.md:41]: suggestion: Both copied and need updating.

**Review by SgtCoDFish — CHANGES_REQUESTED** [review]
I like the changes to the README, but I think the RELEASE file needs updating before it can be merged!

**Comment by inteon** [comment]
> I like the changes to the README, but I think the RELEASE file needs updating before it can be merged!

Thanks, I did blindly copy RELEASE.md, and forgot to update it. Should be fixed now.

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

**Pull request merged**